# **Unit Econimics**


## **1.Objective**


+ This file is used for analyzing and calculating Unit Economics  for the month of March 2023.
+ This is a software company named "Streamline Pro"
+ Compute key performance metrics, including:
  + CAC (Customer Acquisition Cost)
  + ARPU (Average Revenue Per User)
  + COGS (Cost of Goods Sold)
  + Gross Margin
  + LTV (Customer Lifetime Value)
  + LTV/CAC Ratio

## **2.Import data**

You have a shared folder containing company data:
https://drive.google.com/drive/folders/1qhOW9Y2orRXuzbX-kXEmuJ7TMQiRs2Uv?usp=drive_link

In [ ]:
import pandas as pd

In [ ]:
google_sheet_id = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
url='https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
monthly_expenses = pd.read_excel(url, sheet_name='Sheet1')

In [ ]:
google_sheet_id = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
url='https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
payroll = pd.read_excel(url, sheet_name='Sheet1')

In [ ]:
google_sheet_id = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
url='https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
daily_marketing = pd.read_excel(url, sheet_name='Sheet1')

In [ ]:
google_sheet_id = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
url='https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
payments = pd.read_excel(url, sheet_name='Sheet1')

In [ ]:
google_sheet_id = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
url='https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
lifespan = pd.read_excel(url, sheet_name='Sheet1')

## **3.Unit Economics**

### 1.Customer Acquisition Cost(CAC)

Customer Acquisition Cost (CAC) is the total cost a business spends to acquire a new customer

#### 1.1 Software Marketing Expense

Calculate CRM software costs like Salesforce as part of your total expenses, because these tools don’t just serve current customers — they also help turn potential leads into paying ones.


In [ ]:
expense_202303 = monthly_expenses[monthly_expenses['month'] == '2023-03-1']

In [ ]:
crm_expense = expense_202303[expense_202303['item'] == 'Salesforce']['amount'].reset_index(drop=True)[0]
crm_expense

np.int64(1700)

#### 1.2 Salaries

Calculate the personnel salary costs of the Marketing and Sales departments, as these teams are directly involved in acquiring new customers


In [ ]:
payroll_202303 = payroll[payroll['month'] == '2023-03-01']
payroll_marketing_sales = payroll_202303[(payroll_202303['department'] == 'Marketing') | (payroll_202303['department'] == 'Sales')]['paid'].sum()
payroll_marketing_sales

np.int64(5950)

#### 1.3 Daily Marketing Expenses

Compute the total advertising expenses incurred to acquire new customers during the month of March.

In [ ]:
daily_marketing_202303 = daily_marketing[(daily_marketing['date'] >= '2023-03-01') & (daily_marketing['date'] <= '2023-03-31')]
daily_marketing_expense = daily_marketing_202303 ['spending'].sum()
daily_marketing_expense

np.int64(68830)

#### 1.4 Number of Customer

Number of new customers in March

In [ ]:
payments_202303 = payments[(payments['date'] >= '2023-03-01') & (payments['date'] <= '2023-03-31')]

In [ ]:
number_of_new_customer = payments_202303[payments_202303['new_customer'] == 1]['customer_id'].nunique()
number_of_new_customer

63

#### 1.5 CAC Calculation


Customer Acquisition Cost (CAC) is calculated by dividing the total of these costs by the number of new customers.

In [ ]:
cac = (crm_expense + payroll_marketing_sales + daily_marketing_expense) / number_of_new_customer
cac

np.float64(1213.968253968254)

### 2.Average Revenue Per User(ARPU)


ARPU (Average Revenue Per User) is the average revenue generated from each customer over a specific period.


#### 2.1 Total Revenue


Compute the total revenue by aggregating the values in the receipt_amount column of the receipts_history csv.

In [ ]:
total_revenue = payments_202303['receipt_amount'].sum()

#### 2.2 Number of customers

In [ ]:
number_of_customer = payments_202303['customer_id'].nunique()
number_of_customer

292

#### 2.3 ARPU Calculation


In [ ]:
arpu = total_revenue / number_of_customer
arpu

np.float64(284.3595890410959)

### 3.Cost of Goods Sold(COGS)



COGS (Cost of Goods Sold) is the total direct cost incurred to produce or purchase the products sold within a specific period. In this case, since the company operates in the technology sector, COGS is calculated based on the expenses for software used to support business operations.

#### 3.1 Software expenses

Account for the expenses of software tools that are directly involved in the operation of the product system, such as AWS, Google Cloud, and Jira

In [ ]:
production_softwares = ['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira']
production_expense = expense_202303[expense_202303['item'].isin(production_softwares)] ['amount'].sum()
production_expense

np.int64(14200)

Common tools used for internal operations, like Slack and Zoom, should be split with 60% of the cost allocated to COGS and the remaining 40% considered general operating expenses. **Note: This ratio may vary depending on the company.**

In [ ]:
shared_softwares = ['Slack', 'Zoom']
shared_expense = expense_202303 [expense_202303['item'].isin(shared_softwares)] ['amount'].sum() * 0.6

#### 3.2 salaries for Engineering

Salary costs for the engineering team – the people who directly build the product or service.

In [ ]:
production_salary = payroll_202303[payroll_202303['department'] == 'Engineering']['paid'].sum()

#### 3.3 COGS Calculation


In [ ]:
cogs = production_expense + shared_expense + production_salary
cogs

np.float64(20264.0)

### 4.Gross Margin

In [ ]:
gross_margin = (total_revenue - cogs) / total_revenue * 100
gross_margin

np.float64(75.5952452639312)

### 5.Customer Lifetime Value (LTV)

Customer Lifetime Value (LTV) is a metric that measures the total value a customer brings to a business over the entire period they remain a customer.

Calculating the average customer lifespan:
- **lifespan_days**: The number of days each customer stays with the company, from the start date to the churn date.

- **avg_lifespan_days**: The average number of days customers stay with the company.

In [ ]:
lifespan['lifespan_days'] = (lifespan['churn_date'] - lifespan['start_date']).dt.days
avg_lifespan_days = lifespan['lifespan_days'].mean()
avg_lifespan_days

np.float64(295.24)

Convert to months to match the unit of ARPU, which is calculated monthly

In [ ]:
avg_lifespan_month = avg_lifespan_days / 30
avg_lifespan_month

np.float64(9.841333333333333)

In [ ]:
ltv = arpu * avg_lifespan_month * gross_margin
ltv

np.float64(211551.59315068493)

### 6.LTV/CAC

LTV/CAC is a ratio that measures the effectiveness of investment in acquiring customers

In [ ]:
ltv/cac

np.float64(174.26451841648995)

##**4.Summary Report of Financial Metrics March 2023**

### 4.1. Customer Acquisition Cost (CAC)

- **Result:** 1,213.97
- **Comment**: The average Customer Acquisition Cost (CAC) for SaaS businesses typically falls between 200$ and 700$ (**Source: [firstpagesage.com]((https://firstpagesage.com/marketing/average-cac-for-saas-businesses-by-industry-and-customer-type-fc/))**) → Elevated acquisition costs may require strategic optimization.


### 4.2 Average Revenue Per User (ARPU)

- **Result:** 284.36  
- **Comment**:
ARPU is relatively low compared to CAC. However, this can still be suitable for SaaS (subscription-based) models if customers are retained long enough.





### 4.3 Gross Margin

- **Result:** 75.60%
- **Comment:** This is a very high profit margin, which fits well with the software (SaaS) business model. According to KeyBanc and OpenView research, the industry average is around 70–80%


### 4.4 Customer Lifetime Value (LTV)

- **Result**: 211,551.59
- **Comment**: LTV is much higher than CAC, which shows that the value customers bring over their lifetime is very good. This is a sign of a sustainable business model with strong customer retention.



